lightning/src/util/config.rs

```diff
@@ -126,6 +126,26 @@ pub struct ChannelHandshakeConfig {
 	///
 	/// [`KeysInterface::get_shutdown_scriptpubkey`]: crate::chain::keysinterface::KeysInterface::get_shutdown_scriptpubkey
 	pub commit_upfront_shutdown_pubkey: bool,
+
+	/// `their_channel_reserve_satoshis` is the minimum amount that the other node is to keep as a direct payment.
+	/// This ensures that if our counterparty broadcasts a revoked state, we can punish them by
+	/// claiming at least this value on chain
+	///
+	/// `their_channel_reserve_proportion_millionth` is for us to configure counterparty's channel reserve
+	/// for both outgoing and incoming channels.
+	///
+	/// channel_reserve values greater than 30% could be considered highly unreasonable, since that
+	/// amount can never be used for payments.
+	/// Also, if our selected channel_reserve for counterparty and counterparty's selected channel_reserve
+	/// for us, sum upto or greater than channel_value, channel negotiations will fail.
+	///
+	/// `their_channel_reserve_proportion_millionth` is millionths of a channel_value to use as `their_channel_reserve_satoshis`.
+	///
+	/// Default Value is 1% of channel_value, configured as 10,000 millionth
```

Comment:
The text above should mention that any new channels created with the non default value will cause versions of LDK prior to 0.0.104 to refuse to read the `ChannelManager`, similar to how it's mentioned in the explanation for the `max_inbound_htlc_value_in_flight_percent_of_channel` config option.

The reason why that's the case is that `holder_selected_channel_reserve_satoshis` was added with an even number (4) in the tlv fields, and was added in 0.0.104 (though, of course this shouldn't be mentioned in the docs for `their_channel_reserve_proportion_millionth` :) )

Reply:
done 👍